napari/_tests/utils.py

```diff
@@ -140,6 +140,11 @@ def dtype(self) -> DTypeLike:
     def shape(self) -> Tuple[int, ...]:
         return self.data.shape
 
+    @property
+    def ndim(self) -> int:
+        # LayerDataProtocol does not have ndim, but this should be equivalent.
+        return len(self.data.shape)
```

Comment:
I guess this protocol is only really used by images and labels? Otherwise this does not really hold true for layers that accept tuples like surface.

Reply:
Yes. This change was needed because `LayerDataProtocol` also needs `ndim` for `Labels`, so (at most) it only supports `Image` right now.

I like the intent of it and it's often useful for prototyping and testing, but would be great to give it a more thought and care. Outside of this PR though ;)